src/api/procedures/executeManualInstruction.ts

```diff
@@ -70,11 +70,9 @@ export async function prepareExecuteManualInstruction(
   await assertInstructionValidForManualExecution(instructionDetails, context);
 
   if (!portfolios.length) {
-    const {
-      owner: { did: venueOwner },
-    } = await instructionDetails.venue.details();
+    const details = await instructionDetails.venue?.details();
 
-    if (venueOwner !== signerDid && !offChainParties.has(signerDid)) {
+    if (details?.owner.did !== signerDid && !offChainParties.has(signerDid)) {
```

Comment:
```suggestion
    if (!offChainParties.has(signerDid) && (!details || details.owner.did !== signerDid) {
```
If there is no venue associated with the instruction, non-offchain parties should not be allowed in that cases

Reply:
I was looking at the core code but wasn't sure about the logic

Reply:
This is checking for any involvement in the instruction at all, so I don't think this is the place to check for that is we need it.

If there should be no offchain parties then the check would be true in that case anyway so shouldn't really change things here.

Reply:
I think the suggested logic is the same. signerDid should always be defined so we don't need the separate details check. If ?.handles undefined, then it won't equal signer did.